src/exchange/exchange.controller.js

```diff
@@ -64,28 +64,33 @@ angular.module('Module.exchange.controllers').controller(
         this.shouldOpenWizard = false;
         this.hasNoDomain = false;
       });
+    }
+
+    $onInit() {
+      this.services.$scope.resetMessages();
+
+      const modals = {
+        billing: 'exchange/header/update-renew/update-renew',
+        resiliate: 'exchange/header/remove/exchange-remove',
+      };
 
       this.retrievingExchange()
         .then(() => {
-          if ($location.search().action === 'billing') {
-            $timeout(() => {
-              $rootScope.$broadcast(
+          if (Object.keys(modals).includes(this.services.$location.search().action)) {
```

Comment:
You can create a variable such as:

```js
const actionUrlParam = this.services.$location.search().action;
```

And then refer to it:

```suggestion
          if (Object.keys(modals).includes(actionUrlParam)) {
```

Same with [L86](https://github.com/ovh-ux/ovh-module-exchange/pull/209/files#diff-01eeccd5c3e026f4b87e91e822cc21dfR86).